src/RestAPI/dlwsrestapi.py

```diff
@@ -1050,9 +1053,17 @@ def post(self):
                 for i in range(nums[role]):
                     pod_names.append(job_id + "-" + role + str(i))
 
-        # endpoints should be in ["ssh", "ipython"]
-        if any(elem not in ["ssh", "ipython"] for elem in requested_endpoints):
-            return ("Bad request, endpoints only allowed in [\"ssh\", \"ipython\"]: %s" % requested_endpoints), 400
+        interactive_ports = []
+        # endpoints should be ["ssh", "ipython", "tensorboard", {"name": "port name", "podPort": "port on pod in 40000-49999"}]
```

Comment:
Using the same range as ssh_port can cause collision.

Reply:
1. There would be no problem for k8s newtork.
2. For HostNetwork, the range 40000-49999 is for ssh/ipython/tensorboard. When use the HostNetwork, we assume that one pod would take the whole machine, so, the collision is controllable.